segmentstore/server/src/main/java/io/pravega/segmentstore/server/store/StreamSegmentContainerRegistry.java

```diff
@@ -168,6 +186,14 @@ private void unregisterContainer(ContainerWithHandle containerWithHandle) {
         // Notify the handle that the container is now in a Stopped state.
         containerWithHandle.handle.notifyContainerStopped();
         log.info("Unregistered SegmentContainer {}.", containerWithHandle.handle.getContainerId());
+
+        containerWithHandle.shutdownNotifier.complete(null);
+    }
+
+    private static boolean isShutdown(Service.State state) {
+        return state == Service.State.FAILED
+                || state == Service.State.STOPPING
+                || state == Service.State.TERMINATED;
```

Comment:
This is Guava's service state. TERMINATED == Stopped Successfully. FAILED == Stopped with failure.